readthedocs/api/v2/views/footer_views.py

```diff
@@ -216,6 +216,20 @@ def _get_context(self):
         }
         return context
 
+    def is_ad_free_user(self, user):
+        if not settings.USE_PROMOS:
+            return True
+        if user.is_authenticated and hasattr(user, 'gold') and hasattr(user, 'goldonce') and (user.gold.exists() or user.goldonce.exists()):
```

Comment:
I believe the logic is correct. It is an OR except first checking that the attributes exist. Since this API can be called even when the gold app or the donate app are not installed, this extra logic is necessary here.

Reply:
I think Eric is right.

In the case that `hasattr(user, 'gold') -> True` and `user.gold.exists() -> True`, but `hasattr(user, 'goldonce') -> False` it won't enter the IF block but it seems that it's currently ad free.

However, I'm not sure if it's possible that `gold` exists but not `goldonce`, tho.

Maybe this change?

```python
if user.is_authenticated and ((hasattr(user, 'gold') and user.gold.exists()) or (hasattr(user, 'goldonce') and user.goldonce.exists())):
``